utils.py

```diff
@@ -91,6 +92,15 @@ def compute_memory_fragmentation():
 	snapshot = torch.cuda.memory_snapshot()
 	return sum(b['allocated_size'] for b in snapshot) / sum(b['total_size'] for b in snapshot)
 
+
+def compute_memory_fragmentation_ctypes():
```

Comment:
it does not measure memory fragmentation in the allocator. it measures memory utilisation. please rename the function to reflect its semantics, e.g. to `compute_global_device_memory_utilisation`

Reply:
done